include/IGPUBuffer.h

```diff
@@ -30,6 +30,20 @@ class IGPUBuffer : public asset::IBuffer, public IDriverMemoryBacked, public IBa
 
         //! Whether calling updateSubRange will produce any effects.
         virtual bool canUpdateSubRange() const = 0;
+
+        //! A C++11 style move assignment, used for reallocation-like functionality.
+        /** Only ICPUBuffer has a reallocate method, since IGPUBuffer needs an IDriverMemoryAllocation.
+        In the future we may change this API, but the reallocation function would need to either take a
+        memory allocator class and be able to free its current allocation from the old allocator.
+        So if you can provide an already created and allocated IGPUBuffer, we can move its data members
+        from `other` to `this`. However the method does not change the reference count of `other`,
+        the `other` is simply made into an empty IGPUBuffer in a state similar to what it would be if it
+        failed both creation and memory allocation binding. This is why I reccommend that `other` have a
+        reference count of 1 and is not used or bound to any other resource such as a graphics pipeline
+        or a texture buffer object. One can reallocate an IGPUBuffer like this:
+        `{auto rep = Driver->createGPUBufferOnDedMem(newReqs, ... ); A->pseudoMoveAssign(rep); rep->drop();}`
+        \returns true on success, method can fail for a number of reasons such as passing `other==this` .*/
+        virtual bool pseudoMoveAssign(IGPUBuffer* other) = 0;

```

Comment:
not sure I want to support this anymore (would require the update of descriptor sets behind the user's back)

Reply:
the only thing that uses it is somewhere in GPU mem allocators or property pool, cant remember now

Reply:
yeah growable/resizable allocators, we should probably disable them

Reply:
yeah I'm more and more convinced its cancer

Reply:
removed